tests/nightly/gpu/test_bart.py

```diff
@@ -4,35 +4,76 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+import os
 import unittest
+
+from parlai.core.agents import create_agent
+from parlai.core.params import ParlaiParser
 import parlai.utils.testing as testing_utils
+from parlai.utils.testing import tempdir
 
 
 @testing_utils.skipUnlessGPU
 class TestBartModel(unittest.TestCase):
     """
     Test of BART model.
-
-    Checks that BART can be trained on ~ 1k samples of convai2
     """
 
-    @testing_utils.retry(ntries=3, log_retry=True)
     def test_bart(self):
-        valid, test = testing_utils.train_model(
-            dict(
-                task='integration_tests:nocandidate',
-                model='bart',
-                dict_file='zoo:bart/bart_large/model.dict',
-                optimizer='sgd',
-                learningrate=1,
-                batchsize=4,
-                num_epochs=1,
-                short_final_eval=True,
-                validation_max_exs=12,
-            )
+        """
+        Test out-of-the-box BART on repeat task.
+        """
+        valid, _ = testing_utils.eval_model(
+            dict(task='integration_tests', model='bart')
         )
-        self.assertLessEqual(valid['ppl'], 11.0)
-        self.assertLessEqual(test['ppl'], 11.0)
+        self.assertAlmostEqual(valid['ppl'].value(), 1.0, places=1)
+
+    def test_bart_gen(self):
+        """
+        Test out-of-the-box BART generation.
+        """
+        opt = ParlaiParser(True, True).parse_args(['--model', 'bart'])
```

Comment:
you can do `ParlaiParser(True, True).parse_kwargs(model='bart')` if you like that better